File: androidApp/src/main/java/com/dev/kmmwallet/androidApp/viewmodel/LoginViewModel.kt
package com.dev.kmmwallet.androidApp.viewmodel

import com.dev.kmmwallet.shared.datasource.webservice.model.request.User
import com.dev.kmmwallet.shared.viewmodel.event.getUserAuthState
import com.dev.kmmwallet.shared.viewmodel.status.ScreenState

class LoginViewModel : BaseViewModel() {

    fun authenticateUser(
        userName: String,
        pass: String,
        updateUi: (loginState: ScreenState) -> Unit
    ) {
        sharedViewModel.stateProvider.getUserAuthState(
            User(userName, pass),
            fun(loginState: ScreenState) {
                updateUi(loginState)
            })
    }
}

File: iosApp/iosApp/ViewModel/LoginViewModel.swift
//
//  LoginViewModel.swift
//  iosApp
//
//  Created by Chinthaka on 03/04/2021.
//  Copyright © 2021 orgName. All rights reserved.
//

import Foundation
import shared


class LoginViewModel{
    
    private let sharedViewModel: SharedViewModel = SharedViewModel()
    
    public func authenticateUser(user : String , pass : String, completion: @escaping (ScreenState)->()) {
        let user = User(username: user, password: pass)
        self.sharedViewModel.stateProvider.getUserAuthState(user: user) { (state : ScreenState) in
            completion(state)
        }
    }
    
  
}
